drivers/wifi/esp8266/esp8266_offload.c

```diff
@@ -0,0 +1,742 @@
+	}
+
+	k_sem_take(&dev->cmd_handler_data.sem_tx_lock, K_FOREVER);
+	k_sem_reset(&dev->sem_tx_ready);
+
+	ret = modem_cmd_send_nolock(&dev->mctx.iface, &dev->mctx.cmd_handler,
+			     NULL, 0, cmd_buf, &dev->sem_response,
+			     ESP_CMD_TIMEOUT);
+	if (ret < 0) {
+		LOG_DBG("Failed to send command");
+		goto out;
+	}
+
+	ret = modem_cmd_handler_update_cmds(&dev->cmd_handler_data,
+					    cmds, ARRAY_SIZE(cmds),
+					    true);
+	if (ret < 0) {
+		goto out;
+	}
+
+	/*
+	 * After modem handlers have been updated the receive buffer
+	 * needs to be processed again since there might now be a match.
+	 */
+	k_sem_give(&dev->iface_data.rx_sem);
+
+	/* Wait for '>' */
+	ret = k_sem_take(&dev->sem_tx_ready, 5000);
+	if (ret < 0) {
+		LOG_DBG("Timeout waiting for tx");
+		goto out;
+	}
+
+	while (pkt_len) {
+		write_len = MIN(pkt_len, sizeof(dev->pkt_buf));
+		net_pkt_read(sock->tx_pkt, dev->pkt_buf, write_len);
+		dev->mctx.iface.write(&dev->mctx.iface, dev->pkt_buf,
+				      write_len);
+		pkt_len -= write_len;
+	}
+
+	/* Wait for 'SEND OK' or 'SEND FAIL' */
+	k_sem_reset(&dev->sem_response);
+	ret = k_sem_take(&dev->sem_response, ESP_CMD_TIMEOUT);
+	if (ret < 0) {
+		LOG_DBG("No send response");
+		goto out;
+	}
+
+	ret = modem_cmd_handler_get_error(&dev->cmd_handler_data);
+	if (ret != 0) {
+		LOG_DBG("Failed to send data");
+	}
+
+out:
+	(void)modem_cmd_handler_update_cmds(&dev->cmd_handler_data,
+					    NULL, 0U, false);
+	k_sem_give(&dev->cmd_handler_data.sem_tx_lock);
+
+	net_pkt_unref(sock->tx_pkt);
+	sock->tx_pkt = NULL;
+
+	return ret;
+}
+
+static void esp8266_send_work(struct k_work *work)
+{
+	struct esp8266_socket *sock;
+	struct net_context *context;
+	net_context_send_cb_t cb;
+	struct esp8266_data *dev;
+	void *user_data;
+	int ret = 0;
+
+	sock = CONTAINER_OF(work, struct esp8266_socket, send_work);
+	dev = esp8266_socket_to_dev(sock);
+
+	esp8266_lock(dev);
+
+	cb = sock->send_cb;
+	context = sock->context;
+	user_data = sock->send_user_data;
+
+	ret = _sock_send(dev, sock);
+	if (ret < 0) {
+		LOG_ERR("Failed to send data: link %d, ret %d", sock->link_id,
+			ret);
+	}
+
+	esp8266_unlock(dev);
+
+	if (cb) {
+		cb(context, ret, user_data);
+	}
+}
+
+static int esp8266_sendto(struct net_pkt *pkt,
+			  const struct sockaddr *dst_addr,
+			  socklen_t addrlen,
+			  net_context_send_cb_t cb,
+			  s32_t timeout,
+			  void *user_data)
+{
+	struct net_context *context;
+	struct esp8266_socket *sock;
+	struct esp8266_data *dev;
+	int ret = 0;
+
+	context = pkt->context;
+	sock = (struct esp8266_socket *)context->offload_context;
+	dev = esp8266_socket_to_dev(sock);
+
+	LOG_DBG("link %d, timeout %d", sock->link_id, timeout);
+
+	esp8266_lock(dev);
+
+	if (sock->tx_pkt) {
+		esp8266_unlock(dev);
+		return -EBUSY;
+	}
+
+	if (sock->type == SOCK_STREAM) {
+		if (!esp8266_socket_connected(sock)) {
+			esp8266_unlock(dev);
+			return -ENOTCONN;
+		} else if (dst_addr) {
+			esp8266_unlock(dev);
+			return -EISCONN;
```

Comment:
This logic stops the driver working if streaming is invoked, especially with teh changes invoked by **net: mqtt: Add write message handler**
#22794

This code can be removed...to resolve problem

Reply:
Correct, destination address should be ignored for connectable sockets. :+1: